zirgen/Dialect/BigInt/Bytecode/decode.cpp

```diff
@@ -69,6 +70,10 @@ BigIntType Decoder::type(const Op& op) {
   return types[op.type];
 }
 
+mlir::Value Decoder::getPoly(size_t i) {
```

Comment:
Exposing the polys array directly like this would allow an operation to potentially reference an undefined value. I suggest instead renaming the existing `operand` function to `operandA`, then cloning it to make an `operandB` function, which can perform the range check.